.github/workflows/ci.yml

```diff
@@ -33,12 +33,12 @@ jobs:
         uses: actions/cache@v2
         with:
           path: ${{ env.pythonLocation }}
-          key: ${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements.txt') }}
+          key: ${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements.txt') }}-${{ hashFiles('requirements-dev.txt') }}
       - name: Install dependencies
         if: steps.cache-python-env.outputs.cache-hit != 'true'
         run: |
           python -m pip install --upgrade pip
-          pip install pytest
+          pip install --upgrade --upgrade-strategy eager -r requirements-dev.txt -e .
```

Comment:
Is there a particular reason why having `eager` is a good idea? Is it to validate that tests are green for the latest package versions out there?

Reply:
Yes I will remove `--upgrade-strategy eager`

Reply:
But main issue is as we are caching dependencies it is not upgrading unpinned version libraries hence our code is not being tested on latest lib versions. But anyway we can take it in separate task to update dependency cache